Color palettes

crzdg opened the pull request:
Hi!

I created a python script which creates the palettes of the colors for the different intensity maps.

To run you need `svgwrite` installed.

This should help to maintain the README.
If somebody could add an exporter of the actual lua colors to the python dict used in this script, this would be awesome.
Unfortunately, I do not know any lua beyond nvim config.

Also, I rearrange the README a little bit.

Feel free to add some suggestions!

Addresses,

#15 and #31

Have a look on the new README here: https://github.com/crzdg/gruvbox-baby/tree/svg-palettes

luisiacc commented:
Hi there!

It's amazing to see this, thank you! It looks really good.

One small thing: Currently the color palette is the same for all intensities, so it feels kind of repetitive to add the same palette 4 times, when it's really most of the same colors, only some changes in the background colors.

luisiacc commented on README.md:
```diff
@@ -23,12 +23,6 @@ React with typescript example
 Telescope theme inspired by https://github.com/NvChad/NvChad (You have to enable it via config variable)
 ![telescope-theme](https://user-images.githubusercontent.com/31720261/151669762-1470aa12-b6ff-47c1-a4e9-ec9b37e0eabe.png)
 
-#### Data from images:
-
-Font: **Jetbrains Mono patched with nerd fonts**
-
-Terminal: https://github.com/wez/wezterm
-
```
Let's keep this in here, since people people will ask themselves this questions right after seeing the images, if we put them in the bottom is not clear that this info is from this pictures.

luisiacc replied:
It's true that is not very important, but it is a nice-to-have, since almost all the time(at least myself) people are wondering what font/colors/terminals are been used on the code screenshots they see. I don't see any reason to remove them from here, they are there merely as nice-to-have info.

crzdg commented:
> Hi there!
> 
> It's amazing to see this, thank you! It looks really good.
> 
> One small thing: Currently the color palette is the same for all intensities, so it feels kind of repetitive to add the same palette 4 times, when it's really most of the same colors, only some changes in the background colors.

I thought the same here as well. In the end, I decided to add all the images to the `README`, as they all have different background colors (the actual users `background`) so you see the color palette in contrast to the actual `background`-color. IMHO, this is important for the user so the palettes actually can be compared. I agree that most colors are repeated, but for me, the contrast of the colors to the background is an important piece of information.

Maybe we can agree on using an `expandable` (https://gist.github.com/pierrejoubert73/902cc94d79424356a8d20be2b382e1ab) for the different images?

Also, I was thinking of adding an image that shows the differences between the actually changed colors (`dark`, `background`, etc...). So the user can see what shades are used in the different modes.

So in the end the README will contain an image of the medium color palette in expanded mode and an image showing the different shades side-by-side. The palettes for `dark`, `soft`, `soft_flat` will be added as well but are collapsed by default.

What do you think?

luisiacc commented:
Hi there, sry for the late response, I'm traveling so I don't get much time with internet.

> I agree that most colors are repeated, but for me, the contrast of the colors to the background is an important piece of information.

Yeah I agree that the contrast of the colors with the background is important, however that's visible when the colors are seen as code(like on a code screenshot), when you put them as rectangles like the way it is now, that contrast is not really appreciated(At least I didn't see any noticeable difference), after all, the objective of showing the palette is just a way for the user to be able to quickly copy/paste them, therefore having them repeated doesn't make much sense IMO.

> Also, I was thinking of adding an image that shows the differences between the actually changed colors (dark, background, etc...). So the user can see what shades are used in the different modes.
> So in the end the README will contain an image of the medium color palette in expanded mode and an image showing the different shades side-by-side. The palettes for dark, soft, soft_flat will be added as well but are collapsed by default.

OK this sounds good!, although I remain a bit skeptical about having the colors repeated, I love the idea of having the different shades side-by-side.

crzdg commented:
Hey,

No problem.

I will push a suggestion over the weekend. Then we can have the next discussion.
Maybe, we will improve it a little now, and then we do an iteration on it.

mrdgo commented:
Shouldn't you upload media files somewhere else than github?
As a user of this colorscheme, I would like to not have to have the `.svg` files on my machine.

luisiacc commented:
> As a user of this colorscheme, I would like to not have to have the .svg files on my machine.

They are 8kbs each, so you wouldn't even notice them.

crzdg commented:
@luisiacc 

Sorry I was sick the last days and was not able to update this earlier.

OK, I changed quite some things with the new commits.

- First of all, I added a `.gitignore`. I copied the content of the GitHub Lua and Python `.gitignore`. I hope this is OK with you?
- In the `README` I added emojis for all the headers. I also moved back the info about the terminal and the font back to the `Pretty Pictures` section. But instead below an separate header I added it right after the `Pretty Pictures` header.

The main change still would be the `Palette` section.
I added as you have already seen the different palettes of the intensities. But put them in a expandable.
Further, a image showing the different shades of the intensities, as discussed.
And a table with the colors, its names and hex-values.

Have a look at the README here,

https://github.com/crzdg/gruvbox-baby/tree/svg-palettes

I guess we should discuss, how the added content should be organized. The linked `README` shows my proposal for it.
I'm happy to heare your idea of how it should be organized.

luisiacc reviewed (approved):
> Sorry I was sick the last days and was not able to update this earlier.

Hey no worries, hope that you're fine now!

I see this very good now, I think we can merge it.
